x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/action.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AlertData } from '../../../../../common/types';
+
+interface ServerReturnedAlertsData {
```

Comment:
this would be a better place for `Immutable` since the code relies on the fact that action payloads aren't mutated. as it is, the `payload` array is mutable.

```ts
// maybe
interface ServerReturnedAlertsData {  
  readonly type: 'serverReturnedAlertsData';
  readonly payload: readonly AlertData[];
}

// or
type ServerReturnedAlertsData = Immutable<{  
  type: 'serverReturnedAlertsData';
  payload: AlertData[];
}>;
```

Reply:
done